aspnetcore/client-side/dotnet-interop/index.md

```diff
@@ -0,0 +1,913 @@
+{
+    public static async Task Run()
+    {
+        // Ensure JS ES6 module loaded.
+        await JSHost.ImportAsync("PrimitivesShim", "/PrimitivesShim.js");
+
+        // Call a proxy to a static JS method, console.log().
+        PrimitivesInterop.ConsoleLog("Printed from JSImport of console.log()");
+
+        // Basic examples of JS interop with an integer.
+        PrimitivesInterop.IncrementCounter();
+        int counterValue = PrimitivesInterop.GetCounter();
+        PrimitivesInterop.LogInt(counterValue);
+        PrimitivesInterop.LogString("I'm a string from .NET in your browser!");
+
+        // Mapping some other .NET types to JS primitives.
+        PrimitivesInterop.LogValueAndType(true);
+        PrimitivesInterop.LogValueAndType(0x3A); // Byte literal
+        PrimitivesInterop.LogValueAndType('C');
+        PrimitivesInterop.LogValueAndType((Int16)12);
+        // JS Number has a lower max value and can generate overflow errors.
+        PrimitivesInterop.LogValueAndTypeForNumber(9007199254740990L); // Int64/Long
+        // Next line: Int64/Long, JS BigInt supports larger numbers.
+        PrimitivesInterop.LogValueAndTypeForBigInt(1234567890123456789L);// 
+        PrimitivesInterop.LogValueAndType(3.14f); // Single floating point literal
+        PrimitivesInterop.LogValueAndType(3.14d); // Double floating point literal
+        PrimitivesInterop.LogValueAndType("A string");
+    }
+}
+```
+
+The preceding example displays the following output in the browser's debug console:
+
+> :::no-loc text="Printed from JSImport of console.log()":::  
+> :::no-loc text="1":::  
+> :::no-loc text="I'm a string from .NET in your browser!":::  
+> :::no-loc text="boolean true":::  
+> :::no-loc text="number 58":::  
+> :::no-loc text="number 67":::  
+> :::no-loc text="number 12":::  
+> :::no-loc text="number 9007199254740990":::  
+> :::no-loc text="bigint 1234567890123456789n":::  
+> :::no-loc text="number 3.140000104904175":::  
+> :::no-loc text="number 3.14":::  
+> :::no-loc text="string A string":::
+
+## JS `Date` objects
+
+The example in this section demonstrates importing methods which have a [JS `Date`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Date) object as its return or parameter. Dates are marshalled across interop by-value, meaning they are copied in much the same way as JS primitives.
+
+A `Date` object is timezone agnostic. A .NET <xref:System.DateTime> is adjusted relative to its <xref:System.DateTimeKind> when marshalled to a `Date`, but timezone information isn't preserved. Consider initializing a <xref:System.DateTime> with a <xref:System.DateTimeKind.Utc?displayProperty=nameWithType> or <xref:System.DateTimeKind.Local?displayProperty=nameWithType> consistent with the value it represents.
+
+`DateShim.js`:
+
+```javascript
+DateShim.IncrementDay = function (date) {
+  date.setDate(date.getDate() + 1);
+  return date;
+};
+
+DateShim.LogValueAndType = (value) => {
```

Comment:
This isn't logging the type as the name implies. Should it just be the `LogValue` method?